server/src/main/java/com/lunchtime/models/Restaurant.java

```diff
@@ -0,0 +1,89 @@
+package com.lunchtime.models;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+import org.hibernate.annotations.Where;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import java.time.Instant;
+
+@Entity
+@Where(clause = "is_deleted = false or is_deleted is NULL")
+@Setter
+@Getter
+public class Restaurant {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @NotBlank
+    @Column(name = "name")
+    private String name;
+
+    @NotBlank
+    @Email
+    @Column(name = "email")
+    private String email;
+
+    @NotBlank
+    @Column(name = "text_address")
+    private String textAddress;
+
+    @Column(name = "website")
+    private String website;
```

Comment:
There is `@URL` annotation for field's validation as a website. But we should take care for its regex: 

_Per default the constraint validator for this constraint uses the java.net.URL constructor to validate the string. This means that a matching protocol handler needs to be available. Handlers for the following protocols are guaranteed to exist within a default JVM - http, https, ftp, file, and jar. See also the Javadoc for URL._  -> [link](https://softwareengineering.stackexchange.com/questions/356540/how-to-represent-and-validate-website-urls-for-a-jpa-entity)

Reply:
Thanks, very useful annotation! But by default it forbids urls like `somelink.domain.com` (without protocol). It can be customizable by regex, as I understand.
As our business process requires be moderator who check restaurant entity before approve. I think this annotation will bring more complexity then advantage. E.g. if this field must be valid there is must to be validation in input in frontend
I also didn't add annotation `@NotBlank` as restaurant may not have website

Reply:
I think we should embody possible complexity to advantage :) Of course we should add regex so that user can see that his web-URL is not valid, but not moderator.

i.e.: coffee-cafe.com (where first '**c'** is not latin one: this can happen when user starts to type and doesn't switch keyboard language. So, when he notices this he backspace his not-latin letters up to 'c', thinking that 'c' on cyrillic and 'c' on latin is the same :D

Reply:
Regular expressions are very powerful feature and very dangerous and complicated. I think we need to review our requirements according to this conversation and will decide what way is best

There are few references about email:
https://habr.com/en/post/175375/
https://habr.com/en/post/274985/

There is RFC about website naming (I didn't read):
https://tools.ietf.org/html/rfc819

Reply:
I could be wrong. I am going to maintain your opinion. It will be the best decision to review our requirements.

When reviewing, let us remember that this "micro" feature can add point to our developer skills but only if we use smart solution in short time:)

Anyway, may be this "micro" feature is not worth this conversation?)

Sorry in advance :)